test/hotspot/jtreg/compiler/vectorapi/VectorCommutativeOperSharingTest.java

```diff
@@ -0,0 +1,576 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.vectorapi;
+
+import compiler.lib.ir_framework.*;
+import compiler.lib.verify.*;
+import java.util.Random;
+import jdk.incubator.vector.*;
+import jdk.test.lib.Utils;
+
+/**
+ * @test
+ * @bug 8342393
+ * @key randomness
+ * @library /test/lib /
+ * @summary Promote vector IR node sharing
+ * @requires (os.simpleArch == "x64")
```

Comment:
Oh, I see you added it here, rather than at each IR rule. It would be nice to have this test also execute on other machines, for additional coverage.

Can you instead simply restrict it with `avx` features? And if that is not enough, and you do need 64bit restriction, then use this:
`applyIfPlatform = {"64-bit", "true"}`

Reply:
> Oh, I see you added it here, rather than at each IR rule. It would be nice to have this test also execute on other machines, for additional coverage.
> 
These tests are only significant from commoning point of view which is checked through IR rule.

Reply:
That would be sad, because we can still profit from all the value verification for other platforms.

And who knows, we may add similar instructions for other platforms as well, and then the value verification would come in really handy.

Reply:
I again iterate a foundational optimizations like GVN its target indipendent and our intent here is to check commoning, hence verifying it for a target which supports all the IR nodes covered by test should suffice for now.

Reply:
Things might look target independent now, but we are not just writing tests for the state of now, but for the future too. And that is why increased coverage would be good, as I said above ;)